### PR TITLE
Changed strain_seed_value to strain_seed_rate

### PR DESCRIPTION
**Pull request by RaphaelS1**
* `strain_seed_value` set daily seeding to new strain, now `strain_seed_rate` simulated from Poisson(strain_seed_rate). Rate is again divided equally over steps
* https://github.com/mrc-ide/sircovid/issues/245 (point 5)
* tolerance in "calculate Rt with both second variant and vaccination" of [test-carehomes-multistrain.R](https://github.com/mrc-ide/sircovid/blob/6c6a6620ea357da72904a00ec0c95baf34c2a7c4/tests/testthat/test-carehomes-multistrain.R#L745-L747) now set to 0.2 (previously failed as one particle just outside range)
* now uses dust 0.8.10

### CODE REVIEW

**Review by annecori — CHANGES_REQUESTED** [review]
All looks good except minor comment on documentation. Eventually we may want to change the dates of seeding so you only have to specify one date per rate, in the same way we have the beta_values and beta_dates specified, but we should discuss and this can be done in a separate PR I think

**Comment by richfitz** [comment]
I've put a fix for the GPU compilation on 9c1553f, which uses https://github.com/mrc-ide/dust/pull/223; once that version of dust is merged the branch pin can be removed but as this does not affect anything for you you can merge this PR before the dust one is merged

File: R/carehomes.R [hunk 1]
```diff
@@ -872,31 +874,37 @@ carehomes_parameters_strain <- function(strain_transmission, strain_seed_date,
...
+        i <- length(strain_seed_step)
+      } else {
+        i <- seq.int(
+          strain_seed_date[[j]] / dt,
+          strain_seed_date[[j + 1]] / dt - 1
+        )
+      }
+      strain_seed_step[i] <- strain_seed_rate[[j]] * dt
     }
-    assert_integer(strain_seed_value)
```
Comment (annecori) [inline R/carehomes.R:893]: we should probably replace this by a check that the values in there are all positive?

Comment (annecori) [reply R/carehomes.R:893]: and add corresponding tests

File: tests/testthat/test-carehomes-multistrain.R [hunk 2]
```diff
@@ -45,44 +45,50 @@ test_that("carehomes_parameters_strain works as expected", {
...
   expect_equal(sircovid_date_as_date(length(p$strain_seed_step) / 4),
                as.Date("2020-03-02"))
 })
 
 
-test_that("Can seed with multiple-day window", {
-  date <- c("2020-03-01", "2020-03-10")
-  value <- 100
-  p <- carehomes_parameters_strain(c(1, 1), sircovid_date(date), value, 1 / 4)
+test_that("Can seed with two-day window", {
```
Comment (annecori) [inline tests/testthat/test-carehomes-multistrain.R:69]: should this say 10-day?

File: tests/testthat/test-carehomes-multistrain.R [hunk 3]
```diff
@@ -45,44 +45,50 @@ test_that("carehomes_parameters_strain works as expected", {
...
+  rate <- c(100, 0)
+  p <- carehomes_parameters_strain(c(1, 1), sircovid_date(date), rate, 1 / 4)
   expect_equal(sum(p$strain_seed_step), 100 * 10)
   expect_equal(tail(p$strain_seed_step, 6), c(25, 25, 25, 25, 25, 0))
   expect_equal(sircovid_date_as_date(length(p$strain_seed_step) / 4),
                as.Date("2020-03-11"))
 })
 
 
+test_that("Can seed with > two-day window", {
```
Comment (annecori) [inline tests/testthat/test-carehomes-multistrain.R:80]: suggest changing to: "can seed with > 2 dates"

File: tests/testthat/test-carehomes-multistrain.R [hunk 4]
```diff
@@ -118,8 +124,9 @@ test_that("Seeding of second strain generates an epidemic", {
   p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
                             strain_transmission = c(1, 1),
                             strain_seed_date =
-                              sircovid_date(c(date_seeding, date_seeding)),
-                            strain_seed_value = n_seeded_new_strain_inf)
+                              sircovid_date(c(date_seeding, date_seeding)) +
```
Comment (annecori) [inline tests/testthat/test-carehomes-multistrain.R:127]: could write more simply as c(date_seeding, date_seeding + 1)?

File: tests/testthat/test-carehomes-multistrain.R [hunk 5]
```diff
@@ -207,8 +216,9 @@ test_that("N_tot, N_tot2 and N_tot3 stay constant with second strain", {
                             waning_rate = 1 / 20,
                             strain_transmission = c(1, 1),
                             strain_seed_date =
-                              sircovid_date(c(date_seeding, date_seeding)),
-                            strain_seed_value = n_seeded_new_strain_inf)
+                              sircovid_date(c(date_seeding, date_seeding)) +
```
Comment (annecori) [inline tests/testthat/test-carehomes-multistrain.R:219]: same as above

File: tests/testthat/test-carehomes-multistrain.R [hunk 6]
```diff
@@ -231,8 +241,9 @@ test_that("No infection after seeding of second strain with 0 transmission", {
   p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
                             strain_transmission = c(1, 0),
                             strain_seed_date =
-                              sircovid_date(c(date_seeding, date_seeding)),
-                            strain_seed_value = n_seeded_new_strain_inf)
+                              sircovid_date(c(date_seeding, date_seeding)) +
```
Comment (annecori) [inline tests/testthat/test-carehomes-multistrain.R:244]: again

**Review by annecori — CHANGES_REQUESTED** [review]

**Review by annecori — APPROVED** [review]

**Pull request merged**